AutomatticTracks/src/main/java/com/automattic/android/tracks/crashlogging/CrashLoggingDataProvider.kt

```diff
@@ -65,6 +65,11 @@ interface CrashLoggingDataProvider {
         currentExtras: Map<ExtraKnownKey, String>,
         eventLevel: EventLevel
     ): Map<ExtraKnownKey, String>
+
+    /**
+     * Provides the {@link CrashLogging} with information about the current application state.
+     */
+    fun applicationContextProvider(): Map<String, String>
```

Comment:
Here I also applied idea with `provider` which requests for data for before sending any event. This is with sync with other providers and results with really nice interface of `CrashLogging`